patroni/config.py

```diff
@@ -32,6 +32,20 @@
     'channel_binding'
 )
 
+AUTH_ALLOWED_PARAMETERS_MAPPING = {
+    'user': 'PGUSER',
+    'password': 'PGPASSWORD',
+    'sslmode': 'PGSSLMODE',
+    'sslcert': 'PGSSLCERT',
+    'sslkey': 'PGSSLKEY',
+    'sslpassword': '',
+    'sslrootcert': 'PGSSLROOTCERT',
+    'sslcrl': 'PGSSLCRL',
+    'sslcrldir': 'PGSSLCRLDIR',
+    'gssencmode': 'PGGSSENCMODE',
+    'channel_binding': 'PGCHANNELBINDING'
+}
+
```

Comment:
I wonder if this should be here or in `config_generator.py` as it is only used there. If we intend to keep it here as a "public" variable in the module, I'd say to add that to the docstring.

I guess this PR will be merged before https://github.com/zalando/patroni/pull/2708. In that case, if we decide to keep this variable as it is, I can later rebase that PR and document this variable there.

Reply:
```suggestion
AUTH_ALLOWED_PARAMETERS_MAPPING = {k: f"PG{k.upper().replace('_', '')}" if k != 'sslpassword' else ''
                                   for k in _AUTH_ALLOWED_PARAMETERS}
```
Something else that came to my mind is that we can use dict comprehension above.

Reply:
Hmm, actually that dict comprehension would have a problem with `username`.

Reply:
> Hmm, actually that dict comprehension would have a problem with username.

Exactly 🥲

Reply:
```suggestion
AUTH_ALLOWED_PARAMETERS_MAPPING = {
    k: f"PG{k.upper().replace('_', '')}" if k != 'sslpassword' else ''
    for k in _AUTH_ALLOWED_PARAMETERS + ('user',)
    if k != 'username'
}
```
:laughing: